Lib/test/support/__init__.py

```diff
@@ -3093,6 +3093,21 @@ def __fspath__(self):
             return self.path
 
 
+@functools.total_ordering
+class LargestObject:
```

Comment:
How about making these singletons?

`LargestObject = LargestObject()` and `SmallestObject = SmallestObject()`.

I don't think anyone needs to subclass these and there's no need to instantiate more than one of them.

Similarly we can move `ComparesEqualClass` into this function and make it a singleton.